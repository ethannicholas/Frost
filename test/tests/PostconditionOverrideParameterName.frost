@extendable
class PostconditionOverrideParameterName1 {
    var x:Int

    @extendable
    @post(x >= min & x <= max)
    method setup(min:Int, max:Int) {
        x := (min + max) // 2
        Console.printLine(x)
    }
}

class PostconditionOverrideParameterName2 : PostconditionOverrideParameterName1 {
    var y:Int

    @override
    @postAnd(y >= minimum & y <= maximum)
    method setup(minimum:Int, maximum:Int) {
        y := ((minimum + maximum) // 2).abs
        Console.printLine(y)
    }

    @class
    method main() {
        PostconditionOverrideParameterName1().setup(0, 10)
        PostconditionOverrideParameterName2().setup(0, 20)
        PostconditionOverrideParameterName2().setup(-10, -40)
    }
}