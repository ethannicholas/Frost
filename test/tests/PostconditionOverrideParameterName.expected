5
10
25
assertion failure (PostconditionOverrideParameterName.frost:16): postcondition failure for method anonymous.PostconditionOverrideParameterName2.setup(minimum:frost.core.Int, maximum:frost.core.Int)
